src/utils/errorReporting.ts

```diff
@@ -35,16 +36,99 @@ const reportError = (
   };
 };
 
+type Metadata = {
+  [key: string]: any;
+};
+
 const leaveBreadcrumb = (
   message: string,
-  metadata: { [key: string]: any },
+  metadata: Metadata,
   type: BreadcrumbType
 ) => {
   if (!isProductionBuild()) {
     console.debug({ message, metadata, type });
   } else {
     Bugsnag.leaveBreadcrumb(message, metadata, type);
+    addBreadcrumb(convertToSentryBreadcrumb(message, metadata, type));
+  }
+};
+
+/**
+ * Convert a Bugsnag breadcrumb to Sentry's type.
+ * @param message - a string indicating details about the breadcrumb
+ * @param metadata - additional fields
+ * @param type - the type of Bugsnag breadcrumb to be sent
+ * @returns a Sentry breadcrumb
+ */
+const convertToSentryBreadcrumb = (
+  message: string,
+  metadata: Metadata,
+  type: BreadcrumbType
+): Breadcrumb => ({
+  message,
+  data: convertMetadata(metadata),
+  type: convertBreadcrumbType(type),
+});
+
+// The "type" field for Sentry breadcrumbs is just "string", but we can approximate the types listed here:
+// https://develop.sentry.dev/sdk/event-payloads/breadcrumbs/#breadcrumb-types
+enum SentryBreadcrumb {
+  Default = "default",
+  Debug = "debug",
+  Error = "error",
+  Navigation = "navigation",
+  HTTP = "http",
+  Info = "info",
+  Query = "query",
+  Transaction = "transaction",
+  UI = "ui",
+  User = "user",
+}
+
+/**
+ * Convert a Bugsnag breadcrumb type to one of our custom Sentry types.
+ * @param breadcrumbType - the type of Bugsnag breadcrumb to be sent
+ * @returns a custom Sentry breadcrumb type
+ */
+const convertBreadcrumbType = (
+  breadcrumbType: BreadcrumbType
+): SentryBreadcrumb => {
+  switch (breadcrumbType) {
+    case "error":
+      return SentryBreadcrumb.Error;
+    case "navigation":
```

Comment:
I noticed that the navigation breadcrumb has a particular requirement for its data field. Do we just have to remember to include these fields in the metadata?

<img width="458" alt="Screenshot 2023-07-24 at 11 10 28 AM" src="https://github.com/evergreen-ci/spruce/assets/47064971/91391b87-c6eb-4346-88ee-d12bdd8ba950">